src/main/utils/upload.ts

```diff
@@ -47,6 +48,96 @@ const getFileStream = async (filePath: string) => {
     return readableStream;
 };
 
+const getZipFileStream = async (
+    zip: StreamZip.StreamZipAsync,
+    filePath: string
+) => {
+    const stream = await zip.stream(filePath);
+    stream.pause();
+
+    let chunkToConsume = new Uint8Array(),
+        chunkArray: number[] = [],
+        closed = false;
+
+    let resolveObj: (value?: any) => void, rejectObj: (reason?: any) => void;
+
+    stream.on('data', (chunk: Uint8Array) => {
+        for (const byte of chunk) {
+            chunkArray.push(byte);
+        }
+        if (chunkArray.length >= FILE_STREAM_CHUNK_SIZE) {
+            chunkToConsume = new Uint8Array(
+                chunkArray.slice(0, FILE_STREAM_CHUNK_SIZE)
+            );
+            chunkArray = chunkArray.slice(FILE_STREAM_CHUNK_SIZE);
+            resolveObj();
+        }
+    });
+
+    stream.on('end', () => {
+        closed = true;
+        resolveObj();
+    });
+
+    const resumeDataStream = () => {
+        return new Promise((resolve, reject) => {
+            resolveObj = resolve;
+            rejectObj = reject;
+            if (closed) {
+                resolveObj();
+            } else {
+                stream.resume();
+            }
+        });
+    };
+
+    const readableStream = new ReadableStream<Uint8Array>({
+        async pull(controller) {
+            resumeDataStream().then(() => {
+                if (chunkToConsume.length > 0) {
```

Comment:
we can just return the chunk as the promise response, don't think we need to have this separate variable

Reply:
Yep you are right, `chunkToConsume` is redundant here.

Reply:
ohh okk.. got it , just ensure that we read any data left in the chunkArray after the stream ends